pmda/leaflet.py

```diff
@@ -0,0 +1,281 @@
+        parAtoms = db.from_sequence(arraged_coord,
+                                    npartitions=len(arraged_coord))
+        parAtomsMap = parAtoms.map_partitions(self._find_parcc)
+        Components = parAtomsMap.compute(**scheduler_kwargs)
+
+        # Gather the results and start the reduction. TODO: think if it can go
+        # to the private _reduce method of the based class.
+        result = list(Components)
+        # Create the overall connected components of the graph
+        while len(result) != 0:
+            item1 = result[0]
+            result.pop(0)
+            ind = []
+            for i, item2 in enumerate(Components):
+                if item1.intersection(item2):
+                    item1 = item1.union(item2)
+                    ind.append(i)
+            ind.reverse()
+            [Components.pop(j) for j in ind]
+            Components.append(item1)
+
+        # Change output for and return.
+        indices = [np.sort(list(g)) for g in Components]
+        return indices
+
+    def run(self,
+            start=None,
+            stop=None,
+            step=None,
+            scheduler=None,
+            n_jobs=1,
+            n_blocks=None,
+            cutoff=15.0):
+        """Perform the calculation
+
+        Parameters
+        ----------
+        start : int, optional
+            start frame of analysis
+        stop : int, optional
+            stop frame of analysis
+        step : int, optional
+            number of frames to skip between each analysed frame
+        scheduler : dask scheduler, optional
+            Use dask scheduler, defaults to multiprocessing. This can be used
+            to spread work to a distributed scheduler
+        n_jobs : int, optional
+            number of tasks to start, if `-1` use number of logical cpu cores.
+            This argument will be ignored when the distributed scheduler is
+            used
+        n_blocks : int, optional
+            number of partitions to divide trajectory frame into. If ``None``
+            set equal to sqrt(n_jobs) or number of available workers in
+            scheduler.
+
+        """
+        if scheduler is None:
+            scheduler = multiprocessing
+
+        if n_jobs == -1:
+            n_jobs = cpu_count()
+
+        if n_blocks is None:
+            if scheduler == multiprocessing:
+                n_blocks = n_jobs
+            elif isinstance(scheduler, distributed.Client):
+                n_blocks = len(scheduler.ncores())
+            else:
+                raise ValueError(
+                    "Couldn't guess ideal number of blocks from scheduler."
+                    "Please provide `n_blocks` in call to method.")
+
+        universe = mda.Universe(self._top, self._traj)
+        scheduler_kwargs = {'get': scheduler.get}
+        if scheduler == multiprocessing:
+            scheduler_kwargs['num_workers'] = n_jobs
+        
+        start, stop, step = self._trajectory.check_slice_indices(
```

Comment:
none of the start/stop/step things are needed (unless you want to parallelize over frames)

If not needed, please remove to declutter.

Reply:
I do not parallelize over frames, but there is a `for loop` that goes over the frames

Reply:
ok, I missed that